python/test_dcr_template.py

```diff
@@ -305,22 +296,55 @@ def test_dataId_list(self):
             ref_id = {'visit': id, 'raft': '2,2', 'sensor': '1,1', 'filter': band_ref}
             self.assertEqual(ref_id, dataId[i])
```

Comment:
They test slightly different branches of the code, since in one case the function returns a dict and in the other it returns a list of dicts. I've changed the comments to better reflect the difference.